### ISSUE TITLE
Blazor WASM no hosted with gRPC/Docker service

### ISSUE THREAD

**Issue by Henry64** [grpc/grpc-dotnet#1159]
Hi,
i'd like arrange a solution based on several gRPC services included into a docker container. They are up and running. I've checked well with a standalone console project and their services reply well. i must included CORS auth also ...
now i'd like run these services by a Blazored WASM (no hosted) app ...just for a fun ...but it doesn't work : the errore msg is the same like : Grpc.Core.RpcException: Status(StatusCode="Internal", Detail="Error starting gRPC call. HttpRequestException: TypeError: Failed to fetch", DebugException="System.Net.Http.HttpRequestException: TypeError: Failed to fetch
i've applied your BLAZOR scenario but still continue not working ...
Could you help me ?
(i know that with this scenario need to consider to use an apigateway but nevertheless and  just for a fun this simple scenario composed by Blazor WASM no host + gRPC/Docker must run anyway..)

**Comment by BrennanConroy** [comment]
Can you share network traces when the error occurs?

**Comment by Henry64** [comment]
Hello Brennan, here attached the network snapshot captured from Chrome : hoping help..

+ in addiction,  here below the screenshot of my container/Docker where the gRPC service is installed…I’ve tried to call also with http : 49168 and using Kestrel config …but with the same bad  outcome



 

 

Da: Brennan <notifications@github.com> 
Inviato: lunedì 4 gennaio 2021 23:08
A: grpc/grpc-dotnet <grpc-dotnet@noreply.github.com>
Cc: Henry64 <enrico.triveri@gmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [grpc/grpc-dotnet] Blazor WASM no hosted with gRPC/Docker service (#1159)

 

Can you share network traces when the error occurs?

**Comment by BrennanConroy** [comment]
Email replies don't work for attachments and screenshots. Could you please include those in a reply through the GitHub UI?

**Comment by Henry64** [comment]
ok, here attached two documents



[NetworkView.txt](https://github.com/grpc/grpc-dotnet/files/5781105/NetworkView.txt)
![dockerPic](https://user-images.githubusercontent.com/40539618/103883846-08785e00-50de-11eb-8e71-c2a6432da983.jpg)

**Comment by JamesNK** [comment]
I took a look at the network trace. What I noticed:

1. Your Blazor website (https://localhost:44380) and your gRPC service (https://localhost:49167) are hosted on different ports.
2. Your gRPC is call is failing with error `net::ERR_CONNECTION_CLOSED`.

I believe the error indicates that there isn't a gRPC server at https://localhost:49167.

Double check the app is calling the right port, and the service is hosted at that port. Because this is a cross-domain request, CORS must be setup - https://docs.microsoft.com/en-us/aspnet/core/grpc/browser?view=aspnetcore-5.0#grpc-web-and-cors
